src/ilasm/assembler.cpp

```diff
@@ -126,8 +126,25 @@ mdToken Assembler::ResolveClassRef(mdToken tkResScope, __in __nullterminated con
             }
             return tkRet;
         }
-        else  // needs to be resolved
-            if(!m_fIsMscorlib) tkResScope = GetBaseAsmRef();
+        else
+        { // needs to be resolved
+            if ((TypeFromToken(tkResScope) == mdtAssemblyRef) && !IsNilToken(tkResScope))
```

Comment:
Ok, great.  Two more things:
1) please match the prevailing style of the file where there is no space between `if`/`while` and following `(`.  I hate this style, but it's important to match it.  We should probably add these ilasm and ildasm to the list of things covered by `jit-format`, but that's for another change.
2) please add a regression test somewhere under `tests\src\JIT\Regression` (unless you can find a better place).  You should be able to copy another `.il` test and paste your source into it.